bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/remote/ApiClientImpl.kt

```diff
@@ -0,0 +1,161 @@
+package io.bucketeer.sdk.android.internal.remote
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import io.bucketeer.sdk.android.internal.logd
+import io.bucketeer.sdk.android.internal.model.Event
+import io.bucketeer.sdk.android.internal.model.User
+import io.bucketeer.sdk.android.internal.model.request.GetEvaluationsRequest
+import io.bucketeer.sdk.android.internal.model.request.RegisterEventsRequest
+import io.bucketeer.sdk.android.internal.model.response.ErrorResponse
+import io.bucketeer.sdk.android.internal.model.response.GetEvaluationsResponse
+import io.bucketeer.sdk.android.internal.model.response.RegisterEventsResponse
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import java.util.concurrent.TimeUnit
+
+internal class ApiClientImpl(
+  endpoint: String,
+  private val apiKey: String,
+  private val featureTag: String,
+  private val moshi: Moshi
+) : ApiClient {
+
+  private val endpoint = endpoint.toHttpUrl()
+
+  private val client: OkHttpClient = OkHttpClient.Builder()
+    .build()
+
+  private val errorResponseJsonAdapter: JsonAdapter<ErrorResponse> by lazy {
+    moshi.adapter(ErrorResponse::class.java)
+  }
+
+  override fun fetchEvaluations(
+    user: User,
+    userEvaluationsId: String,
+    timeoutMillis: Long?
+  ): GetEvaluationsResult {
+    val body = GetEvaluationsRequest(
+      tag = featureTag,
+      user = user,
+      user_evaluations_id = userEvaluationsId
+    )
+
+    val request = Request.Builder()
+      .url(
+        endpoint.newBuilder()
+          .addPathSegments("v1/gateway/evaluations")
+          .build()
+      )
+      .applyHeaders()
+      .post(body = body.toJsonRequestBody())
+      .build()
+
+    val actualClient = if (timeoutMillis == null) {
+      client
+    } else {
+      client.newBuilder()
+        .callTimeout(timeoutMillis, TimeUnit.MILLISECONDS)
```

Comment:
Could you also implement the timeout logic for the `registerEvents` API, please?

Reply:
Do you mean the same configurable timeout?
If not, OkHttp client has default configuration for request timeout(10 sec each for both connectionTimeout and readTimeout), so no additional configuration is needed

Reply:
> Do you mean the same configurable timeout?

Sorry for forgetting to mention the timeout setting.
Sometimes when we have a spike on the server, until the autoscaling finishes, some of the requests take more than 10 seconds to be processed.
So I was thinking of increasing the default value to 30 seconds for both `GetEvaluations` and `RegisterEvents`.

Giving a second thought to `RegisterEvents`, there is no need to make this optional. We can just set it to 30 seconds.

Please let me know what you think.

Reply:
Thanks, that sounds reasonable 👍 
I'm going to increase the default timeout setting.

Reply:
[`46ffed5` (#62)](https://github.com/ca-dp/bucketeer-android-sdk/pull/62/commits/46ffed5d2f497893279ae986783bfcf6b0a51c3f)